### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by ellthompson**
This PR introduces typescript into the viewer through the use of webpack awesome-typescript-loader and the `tsconfig.json` file.

All src files are now converted to typescript.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/debug.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
+let debugLayerFront: pc.Layer = null;
+let debugLayerBack: pc.Layer = null;
+
+class DebugLines {
+
+    app: pc.Application;
+
```
Comment (slimbuck) [inline src/debug.ts:7]: not loving the empty lines here and below. is there a particular reason for them? could they be removed?

Comment (ellthompson) [reply src/debug.ts:7]: This comes from the following linting rule:
https://eslint.org/docs/rules/lines-between-class-members

It mentions we can turn off the rule for single line class members (`"exceptAfterSingleLine": true`), perhaps thats the route forward. Keeping it enforced for functions.

Comment (slimbuck) [reply src/debug.ts:7]: I think I'd prefer that. Which would you prefer though, lines or no lines?

Comment (ellthompson) [reply src/debug.ts:7]: I'm for no lines after single line members too, it keeps the format of these typings consistent with their assignment in the constructor.

**Pull request merged**